src/controllers/midi/midicontroller.cpp

```diff
@@ -516,9 +516,9 @@ void MidiController::receive(const QByteArray& data, mixxx::Duration timestamp)
         }
     }
 
-    auto it = m_pMapping->getInputMappings().constFind(mappingKey.key);
-    for (; it != m_pMapping->getInputMappings().constEnd() && it.key() == mappingKey.key; ++it) {
-        processInputMapping(it.value(), data, timestamp);
+    const auto inputMappings = m_pMapping->getInputMappings().values(mappingKey.key);
```

Comment:
values() returns a fresh allocated QList()
https://github.com/qt/qtbase/blob/adc0920c48d812fb1b3e2b7bfc76217a7126a41e/src/corelib/tools/qhash.h#L1088C38-L1088C63
This happens for every message and should be avoided.

Reply:
Damn, I tought the QMultiHash was using them internally and just returned an implicitly-shared copy. I should've checked. 

> @Swiftb0y can we find a solution that does not allocate memory?

I didn't at first, but I just found it now, It's https://doc.qt.io/qt-6/qmultihash.html#equal_range. PR underway.

Reply:
Thank you.